src/plugins/mer/meremulatordevice.cpp

```diff
@@ -453,6 +454,56 @@ MerConnection *MerEmulatorDevice::connection() const
     return m_connection.data();
 }
 
+void MerEmulatorDevice::setPortForwardingRule(const QString& ruleName, quint16 hostPort,
+                                              quint16 emulatorVmPort) const {
+    MerVirtualBoxManager::updatePortForwardingRule(m_connection->virtualMachine(), ruleName,
+                                                   hostPort, emulatorVmPort);
+}
+
+/*!
+ * \brief Checks whether the port forwarding rule with the given host machine port is set on the
+ * emulator virtual machine or not.
+ * \param hostPort Port of the host machine to check.
+ * \return True if the port forwarding rule with the given host port is set on the emulator,
+ *         false — otherwise.
+ */
+bool MerEmulatorDevice::isPortForwardingSet(quint16 hostPort) const {
+    return MerVirtualBoxManager::fetchVirtualMachineInfo(m_connection->virtualMachine())
+            .freePorts.contains(hostPort);
+}
+
+/*!
+ * \brief Retrieves the current emulator virtual machine state (running or not).
+ *
+ * This method is used to get the emulator virtual machine state using
+ * MerVirtualBoxManager::isVirtualMachineRunning() synchronously.
+ *
+ * It is possible to do so using the MerEmulatorDevice::connection()->isVirtualMachineOff() method,
+ * but in case when the emulator VM is already running and the user opens Qt Creator this method
+ * returns 'true' by default that means VM is off although the emulator VM is on.
+ * The VM state value will be changed to real after a few milliseconds (~150)
+ * when the MerConnection::virtualMachineOffChanged() signal will be sent.
+ *
+ * So, this method is useful when it needs to get the emulator VM state at particular moment,
+ * for example, during the plugins initialization, when the 'isVirtualMachineOff' value
+ * of MerConnection is not intialized yet.
```

Comment:
I would avoid blocking Qt Creator startup. Upstream have put a lot of effort to keep QtC startup as fast as possible. You should fix your implementation - connect to MerConnection::virtualMachineOffChanged instead.